### PR TITLE
c-ares: change CMake target to cares

### PR DESCRIPTION
**Pull request by Croydon**
https://gitlab.kitware.com/cmake/cmake/-/blob/ee5893c10eba2353741c0db5b28c2401759c393e/Utilities/cmcurl/CMake/FindCARES.cmake

https://github.com/grpc/grpc/blob/9655f1f6449dca36312489f9c82560c32050af71/cmake/cares.cmake#L43

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/c-ares/all/conanfile.py [hunk 1]
```diff
@@ -72,6 +72,9 @@ def package_info(self):
             self.cpp_info.system_libs.extend(["ws2_32", "Advapi32"])
         elif self.settings.os == "Macos":
             self.cpp_info.system_libs.append("resolv")
+
+        self.cpp_info.names["cmake_find_package"] = "cares"
+        self.cpp_info.names["cmake_find_package_multi"] = "cares"
```
Comment (uilianries) [inline recipes/c-ares/all/conanfile.py:77]:
```suggestion
        self.cpp_info.names["cmake_find_package"] = "CARES"
        self.cpp_info.names["cmake_find_package_multi"] = "CARES"
```

Your reference uses upper case

Comment (Croydon) [reply recipes/c-ares/all/conanfile.py:77]: No, the CMake variable names are always lowercase, but both references state the name actual lowercase

Comment (uilianries) [reply recipes/c-ares/all/conanfile.py:77]: Checking again, the c-ares project already provide target files:

```
-- Installing: c-ares/1.16.1/_/_/package/6af9cc7/lib/cmake/c-ares/c-ares-targets.cmake
-- Installing: c-ares/1.16.1/_/_/package/6af9cc7/lib/cmake/c-ares/c-ares-targets-release.cmake
-- Installing: c-ares/1.16.1/_/_/package/6af9cc7/lib/cmake/c-ares/c-ares-config.cmake
-- Installing: c-ares/1.16.1/_/_/package/6af9cc7/lib/cmake/c-ares/c-ares-config-version.cmake
-- Installing: c-ares/1.16.1/_/_/package/6af9cc7/lib/pkgconfig/libcares.pc
```

Thus, the name for cmake should be `c-ares` as we are using by default, however, for pkgconfig we should update to `cares`, just like your change.

**Comment by SpaceIm** [comment]
Why cares for pkgconfig? Why not libcares?

File: recipes/c-ares/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/c-ares/all/conanfile.py:77]: Why cares for pkgconfig? Why not libcares?

Comment (uilianries) [reply recipes/c-ares/all/conanfile.py:77]: @SpaceIm I didn't get, we are using libcares for pkgconfig, take a look at line 78

**Comment by Croydon** [comment]
Another reference:

<details><summary>lib/cmake/c-ares/c-ares-targets.cmake (click for entire file)</summary>

```
cat /home/cr0ydon/.conan/data/c-ares/1.15.0/_/_/package/19729b9559f3ae196cad45cb2b97468ccb75dcd1/lib/cmake/c-ares/c-ares-targets.cmake 
# Generated by CMake

if("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" LESS 2.5)
   message(FATAL_ERROR "CMake >= 2.6.0 required")
endif()
cmake_policy(PUSH)
cmake_policy(VERSION 2.6)
#----------------------------------------------------------------
# Generated CMake target import file.
#----------------------------------------------------------------

# Commands may need to know the format version.
set(CMAKE_IMPORT_FILE_VERSION 1)

# Protect against multiple inclusion, which would fail when already imported targets are added once more.
set(_targetsDefined)
set(_targetsNotDefined)
set(_expectedTargets)
foreach(_expectedTarget c-ares::cares)
  list(APPEND _expectedTargets ${_expectedTarget})
  if(NOT TARGET ${_expectedTarget})
    list(APPEND _targetsNotDefined ${_expectedTarget})
  endif()
  if(TARGET ${_expectedTarget})
    list(APPEND _targetsDefined ${_expectedTarget})
  endif()
endforeach()
if("${_targetsDefined}" STREQUAL "${_expectedTargets}")
  unset(_targetsDefined)
  unset(_targetsNotDefined)
  unset(_expectedTargets)
  set(CMAKE_IMPORT_FILE_VERSION)
  cmake_policy(POP)
  return()
endif()
if(NOT "${_targetsDefined}" STREQUAL "")
  message(FATAL_ERROR "Some (but not all) targets in this export set were already defined.\nTargets Defined: ${_targetsDefined}\nTargets not yet defined: ${_targetsNotDefined}\n")
endif()
unset(_targetsDefined)
unset(_targetsNotDefined)
unset(_expectedTargets)


# Compute the installation prefix relative to this file.
get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
if(_IMPORT_PREFIX STREQUAL "/")
  set(_IMPORT_PREFIX "")
endif()

# Create imported target c-ares::cares
add_library(c-ares::cares STATIC IMPORTED)

set_target_properties(c-ares::cares PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "CARES_STATICLIB"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "rt"
)

if(CMAKE_VERSION VERSION_LESS 2.8.12)
  message(FATAL_ERROR "This file relies on consumers using CMake 2.8.12 or greater.")
endif()

# Load information for each installed configuration.
get_filename_component(_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
file(GLOB CONFIG_FILES "${_DIR}/c-ares-targets-*.cmake")
foreach(f ${CONFIG_FILES})
  include(${f})
endforeach()

# Cleanup temporary variables.
set(_IMPORT_PREFIX)

# Loop over all imported files and verify that they actually exist
foreach(target ${_IMPORT_CHECK_TARGETS} )
  foreach(file ${_IMPORT_CHECK_FILES_FOR_${target}} )
    if(NOT EXISTS "${file}" )
      message(FATAL_ERROR "The imported target \"${target}\" references the file
   \"${file}\"
but this file does not exist.  Possible reasons include:
* The file was deleted, renamed, or moved to another location.
* An install or uninstall procedure did not complete successfully.
* The installation package was faulty and contained
   \"${CMAKE_CURRENT_LIST_FILE}\"
but not all the files it references.
")
    endif()
  endforeach()
  unset(_IMPORT_CHECK_FILES_FOR_${target})
endforeach()
unset(_IMPORT_CHECK_TARGETS)

# This file does not depend on other imported targets which have
# been exported from the same project but in a separate export set.

# Commands beyond this point should not need to know the version.
set(CMAKE_IMPORT_FILE_VERSION)
cmake_policy(POP)
```
</details>


```
# Create imported target c-ares::cares
add_library(c-ares::cares STATIC IMPORTED)

set_target_properties(c-ares::cares PROPERTIES
```

**Comment by Croydon** [comment]
@danimtb Unexpected error. Help! 😄

**Comment by Croydon** [comment]
> You are trying to use the target "cares", however, your change also changes the

What did you wanted to write? You sentence just ended :see_no_evil:

**Comment by uilianries** [comment]
@Croydon This PR changes the cmake file and the namespace, however, the namespace have to be c-ares::cares and the filename Findc-ares.cmake. Here is the source: https://github.com/c-ares/c-ares/blob/master/c-ares-config.cmake.in

**Comment by Croydon** [comment]
Closing in favour of #2160

**Pull request closed without merging**